Examples/CaseStudies/SwiftUICaseStudies/01-GettingStarted-FocusState.swift

```diff
@@ -0,0 +1,108 @@
+import ComposableArchitecture
+import SwiftUI
+
+private let readMe = """
+  This demonstrates how to make use of SwiftUI's `@FocusState` in the Composable Architecture. \
+  If you tap the "Sign in" button while a field is empty, the focus will be changed to that field.
+  """
+
+struct FocusDemoState: Equatable {
+  var focusedField: Field? = nil
+  var password: String = ""
+  var username: String = ""
+
+  enum Field: String, Hashable {
+    case username, password
+  }
+}
+
+enum FocusDemoAction: Equatable {
+  case binding(BindingAction<FocusDemoState>)
+  case signInButtonTapped
+}
+
+struct FocusDemoEnvironment {}
+
+let focusDemoReducer = Reducer<
+  FocusDemoState,
+  FocusDemoAction,
+  FocusDemoEnvironment
+> { state, action, _ in
+  switch action {
+  case .binding:
+    return .none
+
+  case .signInButtonTapped:
+    if state.username.isEmpty {
+      state.focusedField = .username
+    } else if state.password.isEmpty {
+      state.focusedField = .password
+    }
+    return .none
+  }
+}
+.binding(action: /FocusDemoAction.binding)
+
+struct FocusDemoView: View {
+  let store: Store<FocusDemoState, FocusDemoAction>
+  @FocusState var focusedField: FocusDemoState.Field?
```

Comment:
Do we need some availability checks for Xcode 12.5?

Reply:
good catch, added just now